web/components/modules/sidebar.tsx

```diff
@@ -644,6 +644,8 @@ export const ModuleDetailsSidebar: React.FC<Props> = observer((props) => {
           </div>
 
           <div className="flex w-full flex-col items-center justify-start gap-2 border-t border-custom-border-200 px-1.5 py-5">
+            {/* Accessing link outside the disclosure as mobx is not  considering the children inside Disclosure as part of the component hence not observing their state change*/}
```

Comment:
Isn't there a better API in mobx to force observability?

Reply:
How about passing it to defaultOpen in Disclosure?
If that is not an option, at least move it to line 650 inside Disclosure.

Reply:
we add observer to the class to make it observe all the references of the property. But in this situation, not aware of any at this time. I feel the mobx thinks the children inside Disclosure is not inside this component.

One more way to solve this could be that we separate the disclosure panel into a different component and access the store directly from there (not sure if this will work in this situation).

Reply:
I thin the problem is the Disclosure is closed to start with and its not rendering the component. So this is not added to the list of observables in the first render. 
See if it makes sense  to reference it in defaultOpen, else lets move that html inside Disclosure.

Reply:
That is pretty good actually. Works well in this scenario. Made the changes.